authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerMetadataObjects.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization.ranger;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import java.util.List;
+import org.apache.gravitino.MetadataObject;
+
+/** The helper class for {@link RangerMetadataObject}. */
+public class RangerMetadataObjects {
+  private static final Splitter DOT_SPLITTER = Splitter.on('.');
+
+  private static final Joiner DOT_JOINER = Joiner.on('.');
+
+  private RangerMetadataObjects() {}
+
+  /**
+   * Create the Ranger metadata object with the given names and Gravitino type.
+   *
+   * @param names The names of the Gravitino metadata object
+   * @param type The type of the Gravitino metadata object
+   * @return The created Ranger metadata object
+   */
+  public static RangerMetadataObject of(List<String> names, MetadataObject.Type type) {
+    Preconditions.checkArgument(
+        names != null, "Cannot create a Ranger metadata object with null names");
+    Preconditions.checkArgument(
+        !names.isEmpty(), "Cannot create a Ranger metadata object with no names");
+    Preconditions.checkArgument(
+        names.size() <= 3,
```

Comment:
What if we support four-level names later on? For example, like support Trino, how do we handle this scenario? Also how to represent non-table object, like path?